gamestonk_terminal/portfolio/portfolio_controller.py

```diff
@@ -143,6 +184,247 @@ def call_po(self, _):
+        """Process load command"""
+        path = os.path.dirname(os.path.abspath(__file__))
+        path = os.path.abspath(os.path.join(path, "portfolios"))
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="load",
+            description="Load your portfolio",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            choices=[f for f in listdir(path) if isfile(join(path, f))],
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of file to be saved",
+        )
+        if other_args:
+            if "-n" not in other_args and "-h" not in other_args:
+                other_args.insert(0, "-n")
+
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        self.portfolio = portfolio_model.load_df(ns_parser.name)
+        print("")
+
+    @try_except
+    def call_save(self, other_args: List[str]):
+        """Process save command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="save",
+            description="Save your portfolio",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of file to be saved",
+        )
+        if other_args and "-n" not in other_args and "-h" not in other_args:
+            other_args.insert(0, "-n")
+
+        ns_parser = parse_known_args_and_warn(parser, other_args)
+        if not ns_parser:
+            return
+
+        if (
+            ".csv" not in ns_parser.name
+            and ".xlsx" not in ns_parser.name
+            and ".json" not in ns_parser.name
+        ):
+            print(
+                "Please submit as 'filename.filetype' with filetype being csv, xlsx, or json\n"
+            )
+
+        portfolio_model.save_df(self.portfolio, ns_parser.name)
+
+    def call_show(self, _):
+        """Process show command"""
+        portfolio_view.show_df(self.portfolio, False)
+
+    def call_add(self, other_args: List[str]):
+        """Process add command"""
+        parser = argparse.ArgumentParser(
+            add_help=False,
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+            prog="add",
+            description="Adds an item to your portfolio",
+        )
+        parser.add_argument(
+            "-n",
+            "--name",
+            type=str,
+            dest="name",
+            required="-h" not in other_args,
+            help="Name of item to be added (for example a ticker for a stock)",
+        )
+        parser.add_argument(
+            "-t",
+            "--type",
+            dest="type",
+            type=str,
```

Comment:
We can create a function for the `type` that based on what the user selects, we check that the name exists. E.g. if the user chooses a stock `adasdasda` since we wouldn't find it we would not allow it to go through.

I say this because I'm assuming we'll look into the current price of the security, and if it doesn't exist, it may introduce some problems down the line